docs/user_guide/decoupled_models.md

```diff
@@ -87,10 +87,100 @@ exactly one response per request. Even standard ModelInfer RPC in the GRPC endpo
 does not support decoupled responses. In order to run inference on a decoupled
 model, the client must use the bi-directional streaming RPC. See
 [here](https://github.com/triton-inference-server/common/blob/main/protobuf/grpc_service.proto)
-for more details. The [decoupled_test.py](https://github.com/triton-inference-server/server/blob/main/qa/L0_decoupled/decoupled_test.py) demonstrates
+for more details. The [decoupled_test.py](../../qa/L0_decoupled/decoupled_test.py) demonstrates
 how the gRPC streaming can be used to infer decoupled models.
 
 If using [Triton's in-process C API](../customization_guide/inference_protocols.md#in-process-triton-server-api),
 your application should be cognizant that the callback function you registered with 
 `TRITONSERVER_InferenceRequestSetResponseCallback` can be invoked any number of times,
 each time with a new response. You can take a look at [grpc_server.cc](https://github.com/triton-inference-server/server/blob/main/src/grpc_server.cc)
+
+### Knowing When a Decoupled Inference Request is Complete
+
+A request is considered complete when a response containing the
```

Comment:
At a high level - I don't think we need a full description of the different possibilities here. Could we leave it as "a request is considered complete when a response containing the ... is received from a model backend". 

And then split into two sections, "Client applications using the C API can retrieve this via the ...", "Client applications using the streaming GRPC protocol can retrieve this using response parameters. Decoupled models may not send a response for each request but may send a response after receiving multiple requests. In these cases GRPC client can opt-in to receive an empty final response for each request. By default, empty final responses are not sent to save network traffic."

I believe the implementation for a backend should already be covered as that isn't changing - so the existing documentation on sending flags / setting the response complete - should be sufficient - and we shouldn't need to describe how a backend indicates the end - only how a client can know.

Reply:
done